### PR TITLE
add callback full status update header flag

### PR DESCRIPTION
**Pull request by shotasilagadzetaal**

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ they were mined. Metamorph is responsible for storing the transaction data.
 Callbacker is a very simple microservice that is responsible for sending callbacks to clients when a transaction has
 been accepted by the Bitcoin network. To register a callback, the client must add the `X-CallbackUrl` header to the
 request. The callbacker will then send a POST request to the URL specified in the header, with the transaction ID in
-the body. See the [API documentation](/arc/api.html) for more information.
+the body. In case the client wants to receive all the intermidiate status updates about the transaction `X-FullStatusUpdates` header needs to be set to `true`. See the [API documentation](/arc/api.html) for more information.
```
Comment (boecklim) [inline doc/README.md:53]: Type-o on `intermidiate`. Should be `intermediate`

File: api/arc.yml [hunk 2]
```diff
@@ -864,6 +866,13 @@ components:
         Whether we should skip fee validation or not.
       schema:
         type: boolean
+    fullStatusUpdates:
+      name: X-FullStatusUpdates
+      in: header
+      description: >-
+        Whether we should have full status updates in callback or not.
```
Comment (boecklim) [inline api/arc.yml:873]: We should add here that full status updates means an additional callback for status `SEEN_IN_ORPHAN_MEMPOOL` and `SEEN_ON_NETWORK`

File: doc/README.md [hunk 3]
```diff
@@ -50,7 +50,7 @@ they were mined. Metamorph is responsible for storing the transaction data.
 Callbacker is a very simple microservice that is responsible for sending callbacks to clients when a transaction has
 been accepted by the Bitcoin network. To register a callback, the client must add the `X-CallbackUrl` header to the
 request. The callbacker will then send a POST request to the URL specified in the header, with the transaction ID in
-the body. See the [API documentation](/arc/api.html) for more information.
+the body. In case the client wants to receive all the intermediate status updates about the transaction `X-FullStatusUpdates` header needs to be set to `true`. See the [API documentation](/arc/api.html) for more information.
```
Comment (boecklim) [inline doc/README.md:53]: Also here it should be stated that full status updates means an additional callback for status SEEN_IN_ORPHAN_MEMPOOL and SEEN_ON_NETWORK as it is in fact not for all intermediate status updates

File: metamorph/store/postgresql/postgres.go
[the remark this answers is not in the document]
Comment (shotasilagadzetaal) [reply metamorph/store/postgresql/postgres.go:1]: I think it should work. I have added that field to the structure which is being inserted into the database. Nothing else is needed here ? right? 🤔

[the remark this answers is not in the document]
Comment (boecklim) [reply metamorph/store/postgresql/postgres.go:1]: Yes, you may be right 👍

**Review by boecklim — APPROVED** [review]

**Pull request merged**